File: fix_inner_circle.pl
use strict;
use warnings;

my $file = 'src/pages/Home.tsx';
open my $fh, '<', $file or die $!;
my $content = do { local $/; <$fh> };
close $fh;

$content =~ s/<section className="w-full max-w-4xl mx-auto py-32 px-6 text-center relative">/<section className="w-full py-32 px-6 text-center relative bg-[#000000]">/g;
$content =~ s/<h2 className="text-4xl md:text-5xl text-text-main font-medium mb-6 tracking-\[-0.02em\] leading-\[1.1\]">Get our private market intel.<\/h2>/<h2 className="text-4xl md:text-5xl text-white font-medium mb-6 tracking-[-0.02em] leading-[1.1]">Get our private market intel.<\/h2>/g;
$content =~ s/<p className="text-text-muted font-light text-base md:text-lg mb-12 tracking-\[0.02em\] max-w-2xl md:max-w-3xl mx-auto">We share off-market acquisition targets exclusively with our list once a month.<\/p>/<p className="text-white\/80 font-light text-base md:text-lg mb-12 tracking-[0.02em] max-w-2xl md:max-w-3xl mx-auto">We share off-market acquisition targets exclusively with our list once a month.<\/p>/g;

$content =~ s/bg-\[#FFFFFF\] border border-black\/10 p-0 rounded-none relative z-10 focus-within:border-\[#C8102E\]\/50 transition-colors/bg-[#000000] border border-white\/20 p-0 rounded-none relative z-10 focus-within:border-[#C8102E]\/50 transition-colors/g;

$content =~ s/className="flex-1 bg-transparent text-text-main px-6 py-4 outline-none font-mono text-\[10px\] md:text-\[11px\] font-medium tracking-\[0.3em\] uppercase placeholder:text-text-muted rounded-none"/className="flex-1 bg-transparent text-white px-6 py-4 outline-none font-mono text-[10px] md:text-[11px] font-medium tracking-[0.3em] uppercase placeholder:text-white\/50 rounded-none"/g;

open my $out, '>', $file or die $!;
print $out $content;
close $out;

File: fix_inner_circle.py
import os
import re

path = 'src/pages/Home.tsx'
with open(path, 'r') as f:
    content = f.read()

parts = content.split('{/* F. The Inner Circle (Newsletter) */}')
if len(parts) == 2:
    f = parts[1]
    
    # Section background
    f = f.replace('bg-[#000000]', 'bg-gray-50')
    
    # Text colors
    f = f.replace('text-white font-extrabold', 'text-black font-extrabold')
    f = f.replace('text-white/70', 'text-gray-500')
    
    # Input container
    f = f.replace(
        'bg-black border border-gray-700',
        'bg-white border border-gray-200'
    )
    
    # Input field
    f = f.replace(
        '<input type="email" placeholder="ENTER SECURE EMAIL" className="flex-1 bg-black text-white px-6 py-4 outline-none font-mono text-[10px] md:text-[11px] font-medium tracking-[0.3em] uppercase placeholder:text-white/50 rounded-none border-none" />',
        '<input type="email" placeholder="ENTER SECURE EMAIL" className="flex-1 bg-white text-gray-900 px-6 py-4 outline-none font-mono text-[10px] md:text-[11px] font-medium tracking-[0.3em] uppercase placeholder:text-gray-500 rounded-none border-none" />'
    )
    
    new_content = parts[0] + '{/* F. The Inner Circle (Newsletter) */}' + f
    
    with open(path, 'w') as out:
        out.write(new_content)
    print("Updated Inner Circle section.")
